test/export/StructureDefinitionExporter.test.ts

```diff
@@ -1012,4 +1012,24 @@ describe('StructureDefinitionExporter', () => {
       min: 1
     });
   });
+
+  // No duplicate structure definitions exported
+  it('should not export duplicate structure definitions', () => {
```

Comment:
This is most likely me just misunderstanding something, but how exactly is this testing for duplicate structure definitions? I'm guessing that the `ContainsRule` is supposed to do something behind the scenes that adds a second copy of the profile or extension, but I don't know what that would be exactly.

Reply:
This trying to test that there are _not_ duplicate structure definitions. `ContainsRule` makes it so the profile is using the extension, I believe, which is needed to test that only one copy of the profile and extension are exported. The test sets up a profile and extension that would look like the following: 
```
Profile: Foo
Parent: Patient
* extension contains Bar 0..*

Extension: Bar
Id: bar
```

If you were to create those on master and not use the deduplication that Chris added in as a work around, the final package would include the extension twice, since it would get exported once during the profile export and once on its own. You can't exactly copy and paste this test onto master, but a very similar version of it would fail because the length of the `profileDefs` would be `2`, since it would contain the profile and the extension. The test is trying to check that now the length is `1` since the extension should only get added to the appropriate `extensionDefs` array.

If you feel like there's a clearer way to write this test, let me know, because I want to to be clear what it is actually testing. I can also add some of this explanation as a comment if it was actually helpful.

Reply:
Ah, yes, this makes sense. Thank you for clarifying!